### Урок 2. Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

##### 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз

In [1]:
import numpy as np
from math import factorial

def combinations (n, k):
    return factorial(n) / (factorial(k) * factorial(n -k))

def bernuli (p, n, k):
    return combinations(n, k) * p**(k) * (1 - p)**(n - k)

def puason (lmb, m):
    return lmb**m/factorial(m)*np.exp(-lmb)

$$P = (C_{100}^{85}\cdot{0.8}^{85}\cdot{0.2}^{15})\cdot{100} = 4.81\% $$

In [14]:
P = bernuli (0.8, 100, 85)
print(P * 100)

4.806179370074635


##### 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

2.1   $$P \approx (\frac{\lambda^m}{m!}e^{-\lambda})\cdot{100} \approx (\frac{2^{0}}{0!}e^{-2})\cdot{100} \approx  13.5\%$$


In [15]:
lmb = 5000 * 0.0004
print(lmb)
print(puason(lmb, 0))

2.0
0.1353352832366127


2.2   $$P \approx (\frac{\lambda^m}{m!}e^{-\lambda})\cdot{100} \approx (\frac{2^{2}}{2!}e^{-2})\cdot{100} \approx  27.1\%$$

In [16]:
print(puason(lmb, 2))

0.2706705664732254


##### 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [17]:
P = bernuli (0.5, 144, 70)
print(P * 100)

6.281178035144776


$$P = (C_{144}^{70}\cdot{0.5}^{70}\cdot{0.5}^{74})\cdot{100} = 6.28\% $$

##### 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

4.1 

$$ P = ((\frac{7}{10} \cdot \frac{6}{9}) \cdot (\frac{9}{11}\cdot \frac{8}{10}))\cdot{100} \approx 31\% $$

In [22]:
(7/10*6/9*9/11*8/10)*100

30.545454545454543

4.2   возможно 3 исхода (2 и 0) + (1 и 1) + (0 и 2) 
$$ P = ((\frac{C_{7}^{2}*C_{3}^{0}}{C_{10}^{2}} \cdot \frac{C_{9}^{0}*C_{2}^{2}}{C_{11}^{2}}) + (\frac{C_{7}^{1}*C_{3}^{1}}{C_{10}^{2}} \cdot \frac{C_{9}^{1}*C_{2}^{1}}{C_{11}^{2}}) + (\frac{C_{7}^{0}*C_{3}^{2}}{C_{10}^{2}} \cdot \frac{C_{9}^{2}*C_{2}^{0}}{C_{11}^{2}}))\cdot 100 \approx 20.5\% $$

In [25]:
c1 = combinations (7, 2)
c2 = combinations (3, 0)
c3 = combinations (10, 2)
c4 = combinations (9, 0)
c5 = combinations (2, 2)
c6 = combinations (11, 2)
p1 = (c1*c2/c3)*(c4*c5/c6)
print(p1)

0.008484848484848484


In [27]:
c21 = combinations (7, 1)
c22 = combinations (3, 1)
c23 = combinations (10, 2)
c24 = combinations (9, 1)
c25 = combinations (2, 1)
c26 = combinations (11, 2)
p2 = (c21*c22/c23)*(c24*c25/c26)
print(p2)

0.15272727272727274


In [28]:
c31 = combinations (7, 0)
c32 = combinations (3, 2)
c33 = combinations (10, 2)
c34 = combinations (9, 2)
c35 = combinations (2, 0)
c36 = combinations (11, 2)
p3 = (c31*c32/c33)*(c34*c35/c36)
print(p3)

0.04363636363636363


In [30]:
P = p1 + p2 + p3
print(P)

0.20484848484848486


4.3          $$ P = 1 - P(все черные) = 1 - ((\frac{3}{10} \cdot \frac{2}{9}) \cdot (\frac{2}{11}\cdot \frac{1}{10})) \approx 0.999  $$

In [21]:
1 - 3/10*2/9*2/11*1/10

0.9987878787878788